Modern event pattern

BillWagner opened the pull request:
Fixes #244

This PR relies on PR #246, which creates the placeholders for all the delegate and event articles.

BillWagner committed:
Article discussing modern elements of the event pattern.

BillWagner committed:
Article discussing modern elements of the event pattern.

stevehoag committed:
When to Use Generic Collections

Addressed comments and fixed <> syntax

stevehoag committed:
Hashtable and Dictionary Collection Types

Ported and fixed formatting

stevehoag committed:
Thread-Safe Collections

Addressed comments and fixed <> formatting

stevehoag committed:
How to: Use ForEach to Remove Items in a BlockingCollection

Fixed <> formatting

ghogen commented on docs/languages/csharp/modern-events.md:
```diff
@@ -1,12 +1,126 @@
+copy would be a different copy than the one seen by the file search
+object.
+
+Next, let's consider show this change can be backwards compatible.
```
consider how

ghogen commented on docs/languages/csharp/modern-events.md:
```diff
@@ -1,12 +1,126 @@
+existing event argument types do still derive from `System.EventArgs`.
+Backwards compatibility is one major reason why they will continue
+to derive from `System.EventArgs`. Any existing event subscribers will
+be subscriber to an event that followed the classic pattern.
```
will be subscribers to an event that follows the classic pattern?

ghogen commented on docs/languages/csharp/modern-events.md:
```diff
@@ -1,12 +1,126 @@
+subscribers that call async code. The challenge is described in
+the article on [async and await](async.md). Async methods can
+have a void return type, but that is strongly discouraged. When you
+event subscriber code calls an async method, you have no choice but
```
When your

ghogen commented on docs/languages/csharp/modern-events.md:
```diff
@@ -1,12 +1,126 @@
+of the async handler. Because it does not return a task, there is no
+task that can report the error by entering the faulted state. Because
+the method is async, the method can't simply throw the exception. (The
+calling method has continued execution ecause it is `async`.) The
```
because

BillWagner commented:
responded to feedback, rebased and squashed.

thevole committed:
Update event-pattern.md

Small spelling correction

thevole committed:
Update event-pattern.md

Added missing parameters to example constructor.

cartermp committed:
Merge pull request #393 from thevole/patch-1

Update event-pattern.md

cartermp committed:
Merge pull request #394 from thevole/patch-2

Update event-pattern.md

cartermp commented on docs/languages/csharp/modern-events.md:
```diff
@@ -1,12 +1,126 @@
+The previous article discussed the most common event patterns. .NET
+Core has a more relaxed pattern. In this version, the 
+`EventHandler<TEventArgs>` definition no longer has the constraint that
+`TEventArgs` must be a class derived from System.EventArgs.
```
Mind using the backtick formatting here?  System.EventArgs ---> `System.EventArgs`

cartermp commented on docs/languages/csharp/modern-events.md:
````diff
@@ -1,12 +1,126 @@
+    }  
+}  
+```   
+> Note to reviewers: The code for the sample can be updated once
````
This should be updated now that #305 is merged.

cartermp commented on docs/languages/csharp/modern-events.md:
```diff
@@ -1,12 +1,126 @@
+worker.OnStartWorking += async (sender, eventArgs) =>
+{
+    try {
+        await DoWorkAsync();
```
Missing `}`

cartermp commented:
LGTM modulo feedback.

cartermp committed:
Merge pull request #369 from dotnet/32316b

When to Use Generic Collections

Commit:
Fixing up the primer

* Fixing up the primer

* Responding to PR feedback

stevehoag committed:
Comparisons and Sorts Within Collections

* Comparisons and Sorts Within Collections

Fixed <> formatting

* Fixed missing title

Per @cartermp comment.

stevehoag committed:
Commonly Used Collection Types

* Commonly Used Collection Types

Fixed <> formatting

* Fixed link

Per @cartermp  comment

stevehoag committed:
Sorted Collection Types

* Sorted Collection Types

Ported from MSDN (with formatting fixed)

* Fixed typo and See Also header

Ready to merge

cartermp committed:
Hashtable and Dictionary Collection Types

Hashtable and Dictionary Collection Types

stevehoag committed:
Collections and Data Structures

* Collections and Data Structures

Addressed comments and fixed <> formatting

* Fixed two issues

Per #cartermp comments.

* Updated HashSet description

Per @cartermp comment

stevehoag committed:
Merge pull request #376 from dotnet/32816g

Thread-Safe Collections

stevehoag committed:
Merge pull request #380 from dotnet/32816k

How to: Use ForEach to Remove Items in a BlockingCollection

BillWagner committed:
Respond to reviews.

Also, update the sample code

BillWagner committed:
Merge remote-tracking branch 'origin/Modern-Event-Pattern-244' into Modern-Event-Pattern-244

# Conflicts:
#	docs/languages/csharp/modern-events.md

BillWagner commented:
Updated based on @cartermp and @ghogen feedback.

cartermp commented:
@BillWagner excellent.  Feel free to rebase and merge.  Do you have write access to the repo?

BillWagner commented:
@cartermp I don't have write access yet.

cartermp commented:
@BillWagner you should be good once you accept the dotnet org invitation.

BillWagner commented:
Woot! I can commit now. I feel the power coursing through my veins. :imp:
